File: java/sodium/src/sodium/CellSink.java
package sodium;

public final class CellSink<A> extends Cell<A> {
    public CellSink(A initValue) {
    	super(new StreamSink<A>(), initValue);
    }
    
    public void send(A a)
    {
        ((StreamSink<A>)str).send(a);
    }
}

File: f#/src/SodiumFRP.FSharp/_Root/CellSink.fs
namespace SodiumFRP.FSharp

type CellSink<'T> =
    inherit Cell<'T>
    
    val private cellStreamSink : CellStreamSink<'T>
    
    internal new (initialValue) =
        let cellStreamSink = CellStreamSink()
        { inherit Cell<_>(Behavior<_>(cellStreamSink, initialValue)); cellStreamSink = cellStreamSink }
    
    member internal this.Send a = this.cellStreamSink |> StreamSink.send a

module CellSink =
    let create initialValue = CellSink initialValue
    let createWithCoalesce initialValue coalesce = CellSink (initialValue, coalesce)
    let send a (cellSink : CellSink<'T>) = cellSink.Send a